src/main/java/org/geysermc/discordbot/commands/moderation/BanCommand.java

```diff
@@ -42,12 +47,76 @@
 import java.util.Arrays;
 import java.util.List;
 
-public class BanCommand extends Command {
+public class BanCommand extends SlashCommand {
 
     public BanCommand() {
         this.name = "ban";
         this.hidden = true;
-        this.userPermissions = new Permission[] { Permission.BAN_MEMBERS };
+        this.help = "Ban a user";
+
+        this.userPermissions = new Permission[]{Permission.BAN_MEMBERS};
+        this.botPermissions = new Permission[]{Permission.BAN_MEMBERS};
+
+        this.guildOnly = false;
+        this.options = Arrays.asList(
+                new OptionData(OptionType.USER, "member", "The member to ban").setRequired(true),
+                new OptionData(OptionType.INTEGER, "days", "How many days worth of messages should we purge").setRequired(false),
+                new OptionData(OptionType.BOOLEAN, "silent", "Toggle notifying the user upon banning").setRequired(false),
+                new OptionData(OptionType.STRING, "reason", "Specify a reason for banning").setRequired(false)
+        );
+    }
+
+    @Override
+    protected void execute(SlashCommandEvent event) {
+        // Defer to wait for us to handle the command
+        InteractionHook interactionHook = event.deferReply().complete();
+
+        // Fetch options
+        Member member = event.getOption("member").getAsMember();
+        User user = event.getOption("member").getAsUser();
+        Member moderator = event.getMember();
+        int days = 0;
```

Comment:
```suggestion
        int days = event.getOption("days", 0, OptionMapping::getAsInt);
```

Will remove your if statement